source/resource/qx/tool/cli/templates/loader/loader-node.tmpl.js

```diff
@@ -1,6 +1,22 @@
 (function(){ 
 
   var path = require("path");
+  var module = require("module").Module;
+
+  var appModulePaths = [];
+  var old_nodeModulePaths = module._nodeModulePaths;
+  module._nodeModulePaths = function(from) {
+    var paths = old_nodeModulePaths.call(this, from);
+    paths = paths.concat(appModulePaths);
+    return paths;
+  };
+
+  function addNodePath(dir) {
```

Comment:
isn't this going to be quite expensive during startup?  eg when `allScripts` is quite large?

Reply:
allScripts are only the scripts from urisBefore, so the list is quite small

 var allScripts = l.decodeUris(l.urisBefore, "resourceUri");
 t.loadGobalPaths(allScripts);
 t.loadScriptList(allScripts);